OpenRA.Mods.Common/Traits/Cargo.cs

```diff
@@ -269,33 +269,35 @@ public string VoicePhraseForOrder(Actor self, Order order)
 		public bool HasSpace(int weight) { return totalWeight + reservedWeight + weight <= Info.MaxWeight; }
 		public bool IsEmpty(Actor self) { return cargo.Count == 0; }
 
-		public Actor Peek(Actor self) { return cargo.Peek(); }
+		public Actor Peek(Actor self) { return cargo.Last(); }
 
-		public Actor Unload(Actor self)
+		public Actor Unload(Actor self, Actor passenger = null)
 		{
-			var a = cargo.Pop();
+			passenger = passenger ?? cargo.Last();
+			if (!cargo.Remove(passenger))
+				throw new NullReferenceException("Attempted to unload non existing actor from Cargo!");
```

Comment:
This is just nit-picking now, but I'd prefer a `throw new ArgumentException("Attempted to unload an actor that is not a passenger.");`. It's a bit closer to the truth.

Reply:
yeah sounds better, will push a fix in a second